apps/site/lib/site_web/controllers/schedule/line/diagram_format.ex

```diff
@@ -0,0 +1,121 @@
+defmodule SiteWeb.ScheduleController.Line.DiagramFormat do
+  @moduledoc """
+  Even more helpers for the line diagram, using the outputs from SiteWeb.ScheduleController.Line.DiagramHelpers
+  """
+  alias Alerts.{Alert, Match}
+  alias Stops.RouteStop
+
+  @type line_diagram_stop :: %{
+          alerts: [Alert.t()],
+          route_stop: RouteStop.t(),
+          stop_data: [map]
+        }
+
+  # For each set of disrupted stops, maybe make adjustment to the diagram based on
+  # the nature of the disruption.
+  #   * if there is a shuttle we don't style the final stop
+  #   * if there is a detour or stop/station
+  @spec shift_indices_by_effect([number], [Alerts.Alert.effect()]) :: [number]
+  defp shift_indices_by_effect(indices, [:shuttle | _]) do
+    List.delete_at(indices, -1)
+  end
+
+  defp shift_indices_by_effect(indices, [effect | _])
+       when effect in [:stop_closure, :station_closure, :detour]
+       when hd(indices) > 0 do
+    [hd(indices) - 1] ++ indices
+  end
+
+  defp shift_indices_by_effect(indices, [_ | effects]),
+    do: shift_indices_by_effect(indices, effects)
```

Comment:
Of all the lines Codecov is commenting on this one feels like possibly the one most worth considering adding a quick test for.

Reply:
@arkadyan Rewrote this and added a test for this and many others..